contracts/implementations/Mapper.sol

```diff
@@ -1,11 +1,40 @@
 //SPDX-License-Identifier: BUSL-1.1
 pragma solidity ^0.8.13;
 
+import "../interfaces/INFT.sol";
 import "../libraries/MapperStorage.sol";
 
 /// @title Mapper
-/// @notice Contract to map relationships between nodes
+/// @notice Contract to map relationships between nodes and other contracts
 contract Mapper {
+    event BeneficiarySet(
+        address indexed idProxyAddress,
+        uint256 nodeId,
+        address indexed beneficiary
+    );
+
+    /// @notice Sets the beneficiary associated with the pair idProxy/nodeId
+    /// @dev Only the nodeId owner can set a beneficiary
+    /// @param idProxyAddress The address of the NFT proxy
+    /// @param nodeId The node Id to be associated with the beneficiary
+    /// @param beneficiary The address to be a beneficiary
+    function setBeneficiary(
+        address idProxyAddress,
+        uint256 nodeId,
+        address beneficiary
+    ) external {
+        address nodeOwner = INFT(idProxyAddress).ownerOf(nodeId);
+
+        require(nodeOwner == msg.sender, "Only owner");
+        require(nodeOwner != beneficiary, "Beneficiary cannot be the owner");
```

Comment:
If you want to "clear" the beneficiary, is the advice to pass in `address(0)`? This might be worth noting in the doc.